tests/performance/run_server_performance_test.py

```diff
@@ -369,6 +344,29 @@ def main():
 
     stat = StatManager()
 
+    def finish_test(signum, frame):
+        print('-----> Performance test stops. '
+              'Please wait for stopping all subprocesses. <-----')
+
+        global FINISH
+        FINISH = True
+
+        global PROCESSES
+        for proc in PROCESSES:
+            try:
+                proc.terminate()
+            except OSError:
+                pass
+
+        stat.print_stats(args.output)
+        sys.exit("Performance test has timed out or killed.")
+
+    def f(signum, frame):
```

Comment:
@bruntib why do we handle SIGUSR1 and do nothing here?

Reply:
It has been explained in #1826.